File: src/_cffi_src/openssl/rand.py
# This file is dual licensed under the terms of the Apache License, Version
# 2.0, and the BSD License. See the LICENSE file in the root of this repository
# for complete details.

from __future__ import annotations

INCLUDES = """
#include <openssl/rand.h>
"""

TYPES = """
typedef ... RAND_METHOD;
"""

FUNCTIONS = """
void RAND_add(const void *, int, double);
int RAND_status(void);
int RAND_bytes(unsigned char *, int);
"""

CUSTOMIZATIONS = """
"""

File: src/rust/cryptography-openssl/src/rand.rs
// This file is dual licensed under the terms of the Apache License, Version
// 2.0, and the BSD License. See the LICENSE file in the root of this repository
// for complete details.

use crate::OpenSSLResult;

/// Fill a buffer with random bytes.
pub fn rand_bytes(buf: &mut [u8]) -> OpenSSLResult<()> {
    #[cfg(any(
        CRYPTOGRAPHY_IS_LIBRESSL,
        CRYPTOGRAPHY_IS_BORINGSSL,
        CRYPTOGRAPHY_IS_AWSLC
    ))]
    openssl::rand::rand_bytes(buf)?;

    #[cfg(not(any(
        CRYPTOGRAPHY_IS_LIBRESSL,
        CRYPTOGRAPHY_IS_BORINGSSL,
        CRYPTOGRAPHY_IS_AWSLC
    )))]
    openssl::rand::rand_priv_bytes(buf)?;

    Ok(())
}
